tests/RocksDBEngine/ShaEventListener.cpp

```diff
@@ -120,55 +117,79 @@ struct CFilesSetup {
 /// @brief test actions
 ////////////////////////////////////////////////////////////////////////////////
 
-TEST(RocksDBShaCalculatorThread, sha_a_new_file) {
+TEST(RocksDBShaFileHandlerTest, sha_a_new_file) {
   CFilesSetup s;
 
-  std::string new_sst = s._directory.c_str();
-  new_sst += TRI_DIR_SEPARATOR_CHAR;
-  new_sst += "000042.sst";
+  auto shaFileManager = std::make_shared<arangodb::RocksDBShaFileManager>(s._directory);
+
+  std::string filePath = s._directory;
+  filePath += TRI_DIR_SEPARATOR_CHAR;
+  std::string new_sst = filePath + "000042.sst";
 
   auto ret_val = TRI_WriteFile(new_sst.c_str(), "the quick brown fox", 19);
   EXPECT_EQ(ret_val, TRI_ERROR_NO_ERROR);
 
-  auto [good, hash] = arangodb::RocksDBShaCalculatorThread::shaCalcFile(new_sst.c_str());
-  EXPECT_TRUE(good);
-  EXPECT_EQ(hash, "9ecb36561341d18eb65484e833efea61edc74b84cf5e6ae1b81c63533e25fc8f");
+  arangodb::RocksDBSha256Checksum checksumGenerator(new_sst, shaFileManager);
+  if (TRI_ProcessFile(new_sst.c_str(), [&checksumGenerator](char const* buffer, size_t n) {
+        checksumGenerator.Update(buffer, n);
+        return true;
+      })) {
+    checksumGenerator.Finalize();
+  }
+
+  EXPECT_EQ(checksumGenerator.GetChecksum(),
+            "9ecb36561341d18eb65484e833efea61edc74b84cf5e6ae1b81c63533e25fc8f");
+  EXPECT_TRUE(TRI_ExistsFile(
+      (filePath +
+       "000042.sha."
+       "9ecb36561341d18eb65484e833efea61edc74b84cf5e6ae1b81c63533e25fc8f.hash")
+          .c_str()));
 }
 
-TEST(RocksDBShaCalculatorThread, sha_a_different_new_file) {
+TEST(RocksDBShaFileHandlerTest, write_file_with_checksum) {
   CFilesSetup s;
 
-  std::string new_sst = s._directory.c_str();
-  new_sst += TRI_DIR_SEPARATOR_CHAR;
-  new_sst += "000042.sst";
+  std::string filePath = s._directory;
+  filePath += TRI_DIR_SEPARATOR_CHAR;
+  std::string new_sst = filePath + "000042.sst";
 
   auto ret_val = TRI_WriteFile(new_sst.c_str(), "12345 67890 12345 67890", 23);
   EXPECT_EQ(ret_val, TRI_ERROR_NO_ERROR);
 
-  auto [good, hash] = arangodb::RocksDBShaCalculatorThread::shaCalcFile(new_sst.c_str());
-  EXPECT_TRUE(good);
-  EXPECT_EQ(hash, "e7f5561536b5891e35d6021015d67d5798b3731088b44dcebf6bad03785ac8c2");
+  arangodb::RocksDBShaFileManager shaFileManager(s._directory);
+  EXPECT_TRUE(shaFileManager.writeShaFile(
+      new_sst,
+      "e7f5561536b5891e35d6021015d67d5798b3731088b44dcebf6bad03785ac8c2"));
+
+  EXPECT_TRUE(TRI_ExistsFile(
+      (filePath +
+       "000042.sha."
+       "e7f5561536b5891e35d6021015d67d5798b3731088b44dcebf6bad03785ac8c2.hash")
+          .c_str()));
 }
 
-TEST(RocksDBShaCalculatorThread, sha_a_non_existing_file) {
+/*
+TEST(RocksDBShaFileHandlerTest, write_file_name_too_small) {
```

Comment:
It seems that this test is commented out. What shall we do? Enable it or remove it from the test?

Reply:
I will remove it because it was not a useful test I created